packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,485 @@
+      cy.focused().should('not.exist');
+    });
+
+    it('Can click children inside the FTZ', () => {
+      setProps({});
+
+      // wait for first focus to finish
+      cy.focused().should('have.text', 'first');
+
+      // focus inside the FTZ
+      cy.contains('mid').realClick();
+      cy.focused().should('have.text', 'mid');
+    });
+
+    it('Restores focus to FTZ when clicking outside FTZ', () => {
+      setProps({});
+
+      // wait for first focus to finish to avoid timing issue
+      cy.focused().should('have.text', 'first');
+
+      // try to click on button outside FTZ
+      cy.contains('before').realClick();
+      // it focuses first button inside FTZ instead
+      cy.focused().should('have.text', 'first');
+      // and the click isn't respected
+      cy.get('#buttonClicked').should('have.text', 'clicked ');
+    });
+
+    it('Restores focus to FTZ when programmatically focusing outside FTZ', () => {
+      setProps({});
+
+      // wait for first focus to finish to avoid timing issue
+      cy.focused().should('have.text', 'first');
+
+      cy.contains('after').focus();
+      cy.focused().should('have.text', 'first');
+    });
+
+    it('Allows clicks outside FTZ with isClickableOutsideFocusTrap but restores focus inside', () => {
+      setProps({ isClickableOutsideFocusTrap: true });
+
+      // wait for first focus to finish to avoid timing issue
+      cy.focused().should('have.text', 'first');
+
+      // click the button and verify it worked (the story updates the text when a button is clicked)
+      cy.contains('before').realClick();
+      cy.get('#buttonClicked').should('have.text', 'clicked before');
+
+      // but focus is kept within the FTZ
+      cy.focused().should('have.text', 'first');
+    });
+
+    it('Focuses first element when focus enters FTZ with tab', () => {
+      setProps({ disableFirstFocus: true });
+
+      // Start by programmatically focusing an element outside
+      // (clicking it won't work in this case because that would send focus inside the trap)
+      cy.contains('before').focus();
+      cy.focused().should('have.text', 'before');
+
+      // Tab to send focus to the first bumper, which passes focus on to the first element inside
+      cy.realPress('Tab');
+      cy.focused().should('have.text', 'first');
+    });
+
+    it('Focuses last element when focus enters FTZ with shift+tab', () => {
+      setProps({ disableFirstFocus: true });
+
+      // Start by programmatically focusing an element outside
+      // (clicking it won't work in this case because that would send focus inside the trap)
+      cy.contains('after').focus();
+      cy.focused().should('have.text', 'after');
+
+      // Shift+tab will send focus to the last bumper, which passes focus on to the last element inside
+      cy.realPress(['Shift', 'Tab']);
+      cy.focused().should('have.text', 'last');
+    });
+
+    // TODO: investigate why this intermittently fails and re-enable.
```

Comment:
After more investigation, I think the test is flaky because of bugs or timing issues in the function component conversion. That will be fixed and the test enabled in an upcoming PR (the change is ready to go, but it will be hard to review until this PR goes in).